FloripaPublicTransportation/RouteDetailViewController.swift

```diff
@@ -0,0 +1,195 @@
+//
+//  RouteDetailViewController.swift
+//  FloripaPublicTransportation
+//
+//  Created by Marcelo Gobetti on 2/19/16.
+//  Copyright © 2016 Marcelo Gobetti. All rights reserved.
+//
+
+import UIKit
+
+class RouteDetailViewController: UITableViewController {
+    // MARK: Public properties
+    
+    private var _routeId: Int? // stored property
+    var routeId: Int? { // public computed property
+        get { return _routeId }
+        set {
+            guard newValue != nil && newValue != _routeId else {
+                return
+            }
+            
+            _routeId = newValue
+            
+            guard Connectivity.isConnectedToNetwork() else {
+                // cannot call the alert here because the view controller has no view yet
+                return
+            }
+            
+            self.activityIndicator.startAnimating()
+            self.view.bringSubviewToFront(self.activityIndicator)
+            
+            RestApi.findDeparturesByRouteId(_routeId!) { departures in
```

Comment:
can you explain the reasoning behind using a semaphore inside a setter here? Can you explain?